bash_archiving.sh

```diff
@@ -1,117 +1,124 @@
 #!/bin/bash
-USERS=$(cat /etc/passwd | sed 's/\(^[a-zA-Z0-9]*\).*/\1/g' | tr '\n' ' ') ;
-countUSERS=$(wc -l /etc/passwd | grep -o '[0-9]*');
+USERS=$(sed 's/\(^[a-zA-Z0-9]*\).*/\1/g' /etc/passwd | tr '\n' ' ')
+countUSERS=$(wc -l < /etc/passwd)
 x=0 #Заглушка. проверка наличие юзера
 
-if (($# ==  1 ||  $# == 2)) 
+if (($# ==  1 ||  $# == 2))
 then
 	if (($# == 1))
-	then 
-	for (( index=1; index <= $countUSERS; index++))
-	do
-        word=$(cat /etc/passwd | sed 's/\(^[a-zA-Z0-9]*\).*/\1/g' | tr '\n' ' ' | cut -d " " -f $index) 
-		if [ "$1" == "$word" ];then
-     		x=1 #user exist  
-		
-		cat config | while read line; do
- 		line=$(echo  $line | sed 's/ *//g')
-		
-		IFS=:
-		for i in  $line ; do
-		folder=$(echo "$i" | cut -c 2- 2>/dev/null | tr -d '\n') 
-	
-		date=$( date +"_%d_%m_%y-%H_%M_%S" )
-		
-			if [[ $(find /$folder/ -type f -user $1 2>/dev/null;) ]];then 	
-				mkdir -p "$1_$folder";
-				sudo find /$folder/ -type f -user $1 -exec cp -rp {} "$1_$folder"/ 2>/dev/null \;
-				tar -czf "$1_$folder$date.tgz" "$1_$folder"/;
-				echo -e "\n$folder.tgz contain this files:\n";
-				tar -ztvf "$1_$folder$date.tgz"; rm -rf "$1_$folder" 2>/dev/null ;
-			else
-				continue
-			fi      			
+	then
+		for (( index=1; index <= $countUSERS; index++))
+		do
+			word=$(sed 's/\(^[a-zA-Z0-9]*\).*/\1/g' /etc/passwd | tr '\n' ' ' | cut -d " " -f $index)
+
+			if [ "$1" == "$word" ]
+			then
+				x=1 #user exist
+
+				while read line
+				do
+					line=$(sed 's/ *//g' <<< "$line")
```

Comment:
The use of `<<<` allows for any command (which can read from STDIN) to access the contents of the following variable (or other data, such as the output of command substitution) `$line`.